pkg/server/datastore/datastore.go

```diff
@@ -14,34 +12,35 @@ import (
 type DataStore interface {
 	CreateMember(ctx context.Context, m *common.Member) (*Member, error)
 	CreateRelationship(ctx context.Context, r *common.Relationship) (*Relationship, error)
-	CreateAccessToken(ctx context.Context, t *common.AccessToken, memberID uuid.UUID) (*AccessToken, error)
+	GenerateAccessToken(ctx context.Context, t *common.AccessToken, trustDomain string) (*AccessToken, error)
 }
 
 type AccessToken struct {
-	Token  string
-	Expiry time.Time
+	Token    string
+	Expiry   time.Time
+	MemberID uuid.UUID
 }
 
 type Member struct {
 	ID uuid.UUID
 
 	Name        string
 	TrustDomain string
-	Tokens      []AccessToken
 }
 
 type Relationship struct {
 	ID uuid.UUID
 
-	MemberA uuid.UUID
-	MemberB uuid.UUID
+	TrustDomainA string
+	TrustDomainB string
 }
 
 // TODO: use until an actual DataStore implementation is added.
 
 type MemStore struct {
 	member       map[uuid.UUID]*Member
 	relationship map[uuid.UUID]*Relationship
+	token        []*AccessToken
```

Comment:
não rola mapear via memberID aqui? 
```suggestion
	token        map[uuid.UUID]*AccessToken
```